### ISSUE TITLE
AdjustSize for ForceAtlas2 does not take actual size into account

### ISSUE THREAD

**Issue by jacomyma** [graphology/graphology#404]
That must be a remnant of dev vagaries:
https://github.com/graphology/graphology/blob/master/src/layout-forceatlas2/iterate.js#L20
Anyways, as a consequence, that setting does not act as it is expected to, in particular when nodes do not have the same size.

**Comment by jacomyma** [comment]
Here are a few empirical tests.
TL;DR: AdjustSizes works but needs specific settings to be visible. It is ignored if BarnesHut is on, it needs a higher SlowDown to avoid artifacts (~10x more), and is still not super great.

All those tests are based on the same basic layout pass.
```js
forceAtlas2.assign(g, {iterations: 1000, settings: {
	linLogMode: false,
	outboundAttractionDistribution: false,
	adjustSizes: false,
	edgeWeightInfluence: 0,
	scalingRatio: 1,
	strongGravityMode: true,
	gravity: 0.005,
	slowDown: 10,
	barnesHutOptimize: true,
	barnesHutTheta: 1.2,
}});
```

## Test 1 - just the base layout
![image](https://user-images.githubusercontent.com/158525/175279553-302d461f-5559-4e4e-afe1-6d18329fbb36.png)

## Test 2 - Base layout + same but with adjustSize ON
Almost no difference.
![image](https://user-images.githubusercontent.com/158525/175279621-3fcaa076-644b-49e3-b913-6a095845ea44.png)
```js
forceAtlas2.assign(g, {iterations: 1000, settings: {
	linLogMode: false,
	outboundAttractionDistribution: false,
	adjustSizes: true,
	edgeWeightInfluence: 0,
	scalingRatio: 1,
	strongGravityMode: true,
	gravity: 0.005,
	slowDown: 10,
	barnesHutOptimize: true,
	barnesHutTheta: 1.2,
}});
```
## Test 3 - Base layout + same but with adjustSize ON & slowDown to 1
Almost no difference either (pretty bad)
![image](https://user-images.githubusercontent.com/158525/175279823-f28da6e8-3ea3-4901-994d-cd1c570d7f77.png)

```js
forceAtlas2.assign(g, {iterations: 1000, settings: {
	linLogMode: false,
	outboundAttractionDistribution: false,
	adjustSizes: true,
	edgeWeightInfluence: 0,
	scalingRatio: 1,
	strongGravityMode: true,
	gravity: 0.005,
	slowDown: 10,
	barnesHutOptimize: true,
	barnesHutTheta: 1.2,
}});
```
## Test 4 - Base layout + same but with adjustSize ON & slowDown to 1 & barnesHut OFF
Effective but huge artifacts: the network is exploded
![image](https://user-images.githubusercontent.com/158525/175279995-96c23e6a-f34d-48f2-9fe1-0ee286801be3.png)
````js
forceAtlas2.assign(g, {iterations: 1000, settings: {
	linLogMode: false,
	outboundAttractionDistribution: false,
	adjustSizes: true,
	edgeWeightInfluence: 0,
	scalingRatio: 1,
	strongGravityMode: true,
	gravity: 0.005,
	slowDown: 1,
	barnesHutOptimize: false,
	barnesHutTheta: 1.2,
}});
```
## Test 5 - Base layout + same but with adjustSize ON & slowDown to 10 & barnesHut OFF
Still quite effective but still many artifacts!
![image](https://user-images.githubusercontent.com/158525/175280149-44fbe8fd-07d5-499e-bf56-ac1852ccd5e3.png)
```js
forceAtlas2.assign(g, {iterations: 1000, settings: {
	linLogMode: false,
	outboundAttractionDistribution: false,
	adjustSizes: true,
	edgeWeightInfluence: 0,
	scalingRatio: 1,
	strongGravityMode: true,
	gravity: 0.005,
	slowDown: 10,
	barnesHutOptimize: false,
	barnesHutTheta: 1.2,
}});
```
## Test 6 - Base layout + same but with adjustSize ON & slowDown to 100 & barnesHut OFF
Somehow effective but not that great, but no more artifacts
![image](https://user-images.githubusercontent.com/158525/175280327-fcfffd9e-067e-4a74-b710-d12aba17d494.png)

```js
forceAtlas2.assign(g, {iterations: 1000, settings: {
	linLogMode: false,
	outboundAttractionDistribution: false,
	adjustSizes: true,
	edgeWeightInfluence: 0,
	scalingRatio: 1,
	strongGravityMode: true,
	gravity: 0.005,
	slowDown: 100,
	barnesHutOptimize: false,
	barnesHutTheta: 1.2,
}});
```

## Test 7 - Base layout + same but with adjustSize ON & slowDown to 100 & 10000 iterations & barnesHut ON
Clearly, when BarnesHut is activated, adjustSize is ignored.
![image](https://user-images.githubusercontent.com/158525/175280430-93390335-cd70-4d92-9bbf-b16ccab8db26.png)
```js
forceAtlas2.assign(g, {iterations: 10000, settings: {
	linLogMode: false,
	outboundAttractionDistribution: false,
	adjustSizes: true,
	edgeWeightInfluence: 0,
	scalingRatio: 1,
	strongGravityMode: true,
	gravity: 0.005,
	slowDown: 100,
	barnesHutOptimize: true,
	barnesHutTheta: 1.2,
}});
```

**Comment by cliffordfajardo** [comment]
Hey @jacomyma - do you have a link to your github branch or a codesandbox reproducing the graph's above?
I'm new to using graphology and am learning how to setup library in my browser to render

Any help would be appreciated 🙏

**Comment by jacomyma** [comment]
Not sure whether this is the right place for this but whatever... Force atlas 2 is just an extension to Graphology. Its doc is right there: https://graphology.github.io/standard-library/layout-forceatlas2.html You can render Graphology networks with Sigma directly, here are a bunch of sandboxes: https://codesandbox.io/examples/package/graphology

**Comment by apitts** [comment]
@jacomyma and @Yomguithereal I believe that the two +='s here should be replaced with -='s, i.e. I believe the sign should be reversed: https://github.com/graphology/graphology/blob/master/src/layout-forceatlas2/iterate.js#L527-L528

**Comment by jacomyma** [comment]
@apitts agreed: here is the equivalent line in the Java implementation. It confirms the ```+=``` in Graphology is a typo, and should be ```-=```.
https://github.com/gephi/gephi/blob/master/modules/LayoutPlugin/src/main/java/org/gephi/layout/plugin/forceAtlas2/ForceFactory.java#L218-L219

**Comment by Yomguithereal** [comment]
I have fixed the code as per your discovery and released a v0.10.1 :)
